airsonic-main/src/main/java/org/airsonic/player/dao/AbstractDao.java

```diff
@@ -69,11 +73,19 @@ protected String prefix(String columns, String prefix) {
         l.replaceAll(s -> prefix + "." + s);
         return String.join(", ", l);
     }
+    
+    protected static Object[] convertArgs(Object[] args) {
+        return args == null ? null : Stream.of(args).map(x -> (Object) ((x instanceof Instant) ? Timestamp.from((Instant) x) : x)).collect(Collectors.toList()).toArray();
```

Comment:
Double ternaries are gross, wouldn't it be a bit clearer to split this on different lines?

Reply:
I could use Optionals too, I just think this is cleaner, otherwise we'll have to use block statements with returns inside the `map()`